src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1982,6 +2149,53 @@ def translate_device(dev):
         else:
             return False
 
+    @private
+    def get_iommu_type(self):
+        def check_iommu(iommu_type):
+            IOMMU_TEST = {'VT-d': {'arglist': ['/usr/sbin/acpidump', '-t'],
```

Comment:
@stattin42 this looks good to me, however I was thinking do we really need to run `grep` ? How about just parsing the output in python instead of managing 2 processes and ensuring everything is good wrt their management ?

Reply:
@sonicaj, I have revised to skip grep in favour of parsing the output in python. To further shield middlewared from poential misbehaviour of the external utility and to avoid the need to buffer the complete output in python I also switched to asyncio.